File: ts/src/util/numeric.ts
/**
 * Returns the positive modulus of x mod n.
 *
 * @param x - The number to be reduced.
 * @param n - The modulus.
 *
 * @returns bigint - The positive modulus of x mod n.
 */
function bigModPos(x: bigint, n: bigint): bigint {
    return ((x % n) + n) % n;
}

/**
 * Compares two bigint values.
 *
 * @param a - The first value to compare.
 * @param b - The second value to compare.
 *
 * @returns number - Returns -1 if a < b, 1 if a > b, and 0 if a == b.
 */
function bigCmp(a: bigint, b: bigint): number {
    if (a < b) return -1;
    if (a > b) return 1;
    return 0;
}

export {
    bigModPos,
    bigCmp
};

File: go/util/numeric.go
package util

import "math/big"

// BigModPos returns x modulo n as a non-negative result.
// x is the dividend, n is the modulus. The result is in [0, n).
func BigModPos(x, n *big.Int) *big.Int {
	mod := new(big.Int).Mod(x, n)
	if mod.Sign() < 0 {
		mod.Add(mod, n)
	}
	return mod
}

// BigCmp Compares two big.Int values.
// Returns -1 if a < b, 0 if a == b, and 1 if a > b.
func BigCmp(a, b *big.Int) int {
	return a.Cmp(b)
}
